sycl/source/detail/os_util.cpp

```diff
@@ -233,15 +238,14 @@ void OSUtil::alignedFree(void *Ptr) {
 #endif
 }
 
-/* This is temporary solution until std::filesystem is available when SYCL RT
- * is moved to c++17 standard*/
-
-/* Create directory recursively and return non zero code on success*/
+// throws on error.
 int OSUtil::makeDir(const char *Dir) {
   assert((Dir != nullptr) && "Passed null-pointer as directory name.");
   if (isPathPresent(Dir))
     return 0;
 
+// older GCC doesn't have full C++ 17 support.
+#if __GNUC__ && __GNUC__ < 8
```

Comment:
Should we throw in this case as well, to align the two branches?

Reply:
changing the signature of this function leads to an ABI break. Since we need to keep this old stuff anyway, I was thinking to keep it as is.  
My preference would be to change the signature and drop GCC < 8 support, but that decision isn't mine to make.

Reply:
We can keep the signature as is and still throw when we would have returned a non-zero code. I don't think we're handling non-zero codes returned from this anyway, are we?

Reply:
`std::fileystem::create_directories` throws a `std::filesystem::filesystem_error` but the "old" code  doesn't have `std::filesystem` so can't throw that.    I'm modifying it to throw std::runtime_error